### PR TITLE
Allow using custom keys and certificates with FDO servers

### LINKED ISSUE #16: Allow a user to provide custom keys and certificates for a manufacturing server
##### SUMMARY
It is currently not possible to provide keys and certificates to a manufacturing server as they are generated as part of the setup role. 
This means that:
1. There is no way to use custom keys and certificates, e.g. based on a corporate CA. 
2. The manufacturing server knows the owner's private key (or rather owns it). It looks like FDO was meant for use cases when the manufacturing and the owner are separate entities, especially when considering the late binding feature.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
roles/setup_manufacturing_server

##### ADDITIONAL INFORMATION

The role should be able receive a directory that contains just the keys and certificates a manufacturing server requires. 
* diun_key.der
* diun_cert.pem
* manufacturer_cert.pem
* manufacturer_key.der
* owner_cert.pem
* device_ca_key.der
* device_ca_cert.pem

Where _owner\_cert.pem_ may originate at a device's prospective owner - not the manufacturer.
A conditional or a separate role should be used to optionally generate keys and certificates for demo purposes.

```yaml
- hosts: manufacturing_server
  gather_facts: true
  become: true
  vars:
    localhost_manufacturing_server_certs_dir: "{{ playbook_dir }}/fdo_keys"
  roles:
  - setup_manufacturing_server
```

### PR DESCRIPTION
**Pull request by empovit**
##### SUMMARY

Fixes #16 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
roles/copy_manufacturing_server_certs_to_manufacturing_server
roles/generate_keys_and_certificates
roles/setup_manufacturing_server

##### ADDITIONAL INFORMATION

The changes are backward compatible, meaning that existing playbooks that rely on keys and certificates being generated by the manufacturer will keep behaving exactly as they did.

See the updated _examples/README.md_ and sample playbooks for usage.

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -202,3 +202,70 @@ In case of the specific use-case where Manufacturing Server is located in differ
...
+it is possible to use custom keys and certificates instead of having them
+generated by the manufacturing server. The following examples show how to
+distribute keys and certificates to FDO servers. The collection also includes
+a role for generating all or a subset of keys and certificates if needed.
+
+```yaml
+---
+- name: Generate FDO Keys and Certificates
+  hosts: localhost
+  become: true
```
Comment (fabiendupont) [inline examples/README.md:218]: Since the default destination is `/var/tmp/fdo` why do we need privilege escalation?

**Review by fabiendupont — COMMENTED** [review]
What about the case when the users have generated the keys and certificates out-of-band, so are not using the `generate_keys_and_certificates` role? Could you document how they should structure the folder where they put the certificates and keys, so that they can be install by the other roles?

File: examples/README.md (hunk 1 above)
Comment (empovit) [reply examples/README.md:218]: @fabiendupont for installing `fdo-admin-cli` if it wasn't installed. All these roles might or might not require `become: true` depending on the Ansible user, it's just an example.

Comment (empovit) [reply examples/README.md:218]: Regarding out-of-band generation, you'll need to look at the role. The examples cover only basic scenarios and I didn't want to change that for now. It's also no different than before for rendezvous and owner, what's changed is the manufacturer, but I kept the same pattern. In any case, there's a lot of documenting to do later.

Comment (empovit) [reply examples/README.md:218]: Actually, it could be a good idea to at least point to the relevant section in the doc. Will do that.

**Review by maxamillion — APPROVED** [review]
This is awesome! Thank you for the PR!

**Review by fabiendupont — APPROVED** [review]

**Comment by fabiendupont** [comment]
Looks good to me.

**Pull request merged**